DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -9,131 +10,146 @@
+            A.CallTo(() => sessionDataService.StartOrRestartDelegateSession(A<int>._, A<int>._)).MustNotHaveHappened();
         }
 
         [Test]
-        public void StartOrUpdateSession_should_not_modify_context_for_course_in_session()
+        public void StartOrUpdateDelegateSession_should_not_modify_context_for_course_in_session()
         {
             // Given
             httpContextSession.Clear();
             const int courseInSession = CustomisationId;
             httpContextSession.SetInt32($"SessionID-{courseInSession}", DefaultSessionId);
 
             // When
-            sessionService.StartOrUpdateSession(CandidateId, courseInSession, httpContextSession);
+            sessionService.StartOrUpdateDelegateSession(CandidateId, courseInSession, httpContextSession);
 
             // Then
             httpContextSession.Keys.Should().BeEquivalentTo($"SessionID-{courseInSession}");
             httpContextSession.GetInt32($"SessionID-{courseInSession}").Should().Be(DefaultSessionId);
         }
 
         [Test]
-        public void StopSession_should_close_sessions()
+        public void StopDelegateSession_should_close_sessions()
         {
             // Given
             httpContextSession.Clear();
             const int courseInSession = CustomisationId;
             httpContextSession.SetInt32($"SessionID-{courseInSession}", DefaultSessionId);
 
             // When
-            sessionService.StopSession(CandidateId, httpContextSession);
+            sessionService.StopDelegateSession(CandidateId, httpContextSession);
 
             // Then
-            A.CallTo(() => sessionDataService.StopSession(CandidateId)).MustHaveHappenedOnceExactly();
-            A.CallTo(() => sessionDataService.StopSession(A<int>._))
+            A.CallTo(() => sessionDataService.StopDelegateSession(CandidateId)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => sessionDataService.StopDelegateSession(A<int>._))
                 .WhenArgumentsMatch((int candidateId) => candidateId != CandidateId)
                 .MustNotHaveHappened();
         }
 
         [Test]
-        public void StopSession_should_clear_context()
+        public void StopDelegateSession_should_clear_context()
         {
             // Given
             httpContextSession.Clear();
             const int courseInSession = CustomisationId;
             httpContextSession.SetInt32($"SessionID-{courseInSession}", DefaultSessionId);
 
             // When
-            sessionService.StopSession(CandidateId, httpContextSession);
+            sessionService.StopDelegateSession(CandidateId, httpContextSession);
 
             // Then
             httpContextSession.Keys.Should().BeEmpty();
         }
+
+        [Test]
+        public void StartAdminSession_should_call_data_service()
+        {
+            // Given
+            A.CallTo(() => sessionDataService.StartAdminSession(A<int>._)).Returns(1);

```

Comment:
Could we make this test slightly more accurate by only returning "1" if the "sessionDataService.StartAdminSession" is called with the right param:
`A.CallTo(() => sessionDataService.StartAdminSession(1)).Returns(1);`

Reply:
Oh, ignore this, we check it below, so it actually doesn't matter 😄